mlrun/model_monitoring/db/stores/v3io_kv/kv_store.py

```diff
@@ -703,3 +702,57 @@ def get_v3io_monitoring_apps_container(project_name: str) -> str:
     @staticmethod
     def _get_monitoring_schedules_container(project_name: str) -> str:
         return f"users/pipelines/{project_name}/monitoring-schedules/functions"
+
+    @staticmethod
+    def _extract_metrics_from_items(
+        items: list[dict[str, str]],
+    ) -> list[mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric]:
+        metrics: list[
+            mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric
+        ] = []
+        for app_item in items:
+            # See https://www.iguazio.com/docs/latest-release/services/data-layer/reference/system-attributes/#sys-attr-__name
+            app_name = app_item.pop("__name")
+            if app_name == ".#schema":
+                continue
+            for result_name in app_item:
+                metrics.append(
+                    mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric(
+                        app=app_name,
+                        name=result_name,
+                        type=mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetricType.RESULT,
+                        full_name=f"{app_name}/{mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetricType.RESULT}/{result_name}",
+                    )
+                )
+        return metrics
+
+    def get_model_endpoint_metrics(
+        self, endpoint_id: str
+    ) -> list[mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric]:
+        """Get model monitoring results and metrics on the endpoint"""
+        metrics: list[
+            mlrun.common.schemas.model_monitoring.ModelEndpointMonitoringMetric
+        ] = []
+        container = self.get_v3io_monitoring_apps_container(self.project)
+        try:
+            response = self.client.kv.scan(container=container, table_path=endpoint_id)
+        except v3io.dataplane.response.HttpResponseError:
+            logger.info(
+                "Attempt getting metrics and results - no data. Check the "
+                "project name, endpoint, or wait for the applications to start.",
+                container=container,
+                table_path=endpoint_id,
+            )
+            return []
+
+        while response.output.items:
```

Comment:
```suggestion
        while True:
```
As we discussed, you can't make this assumption.

Reply:
IIRC we talked about the `.last` attribute.
I assume that if the items are empty, there is nothing left. Am I wrong?
I also tried to avoid `while True`, as it's a bit scary

Reply:
And I said you can't make this assumption 😉

Reply:
Ok, `while True` it is